### PR TITLE
SAF-1266 Extend event-sourcing lib

### PR DESCRIPTION
**Pull request by sheremet**
## SAF-1266 Extend event-sourcing lib to pass to Event Handler reference to AggregateRootAsync used in generation of event

- upgrade lib dependencies
- UpdateEventHandlers get reference to AggregateRootAsync emitted event to be able to optimize persistency.
- fix linter warnings

Signed-off-by: Ruslan Sheremet <info.webstudio@gmail.com>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,40 +12,44 @@
     "prepublish:next": "npm run build",
     "publish:next": "npm publish --access public --tag next",
     "lint": "eslint \"{src,test}/**/*.ts\" --fix && npm run prettier",
-    "prettier": "prettier --write 'src/*.{ts, js}'"
+    "prettier": "prettier --write \"src/**/*.ts\"",
+    "deps-update": "ncu --packageFile package.json -i"
```
Comment (spunkypanda) [inline package.json]: Maybe we should use npx to use packages which are expected to be globally installed.
`npx npm-check-updates  --packageFile package.json -i`

File: src/view/view-updater.ts [hunk 2]
```diff
@@ -1,27 +1,37 @@
...
+    const updater = ViewUpdaters.get(event.constructor.name);
+    if (updater) {
+      if (!this.instances.has(updater)) {
+        this.instances.set(
+          updater,
+          this.moduleRef.get(updater.name, { strict: false }),
+        );
+      }
+      if (rootAggregator !== undefined) {
+        await this.instances.get(updater).handle(event, rootAggregator);
```
Comment (spunkypanda) [inline src/view/view-updater.ts:30]: @sheremet  Handler functions currently implement the `IEventHandler` interface from `@nestjs/cqrs`, which doesn't tell the developer that there's a second argument that can be passed to the handle method.
Maybe, we need to create and export an additional interface that takes this into account.

Screen Shot 2020-11-05 at 4 42 33 PM

Comment (sheremet) [reply src/view/view-updater.ts:30]: @spunkypanda In this place handle method from IViewUpdater interface, that extended for using `rootAggregator`

File: package.json (hunk 1 above)
Comment (sheremet) [reply package.json]: Updated, please check

File: src/eventstore.ts [hunk 3]
```diff
@@ -1,34 +1,46 @@
...
+  }
 
-    this.eventstore.init(err => {
-      if (err) {
-        throw err;
-      }
-      this.eventStoreLaunched = true;
-    });
+  private initEventStore(config: DatabaseConfig) {
+    switch (config.dialect) {
```
Comment (bobchainer) [inline src/eventstore.ts:20]: we are using elasticsearch & oracle with eventstore lib

File: src/store-event-bus.ts [hunk 4]
```diff
@@ -1,17 +1,21 @@
...
 import { EventStore } from './eventstore';
-import { StorableEvent } from './interfaces/storable-event';
-import { ViewEventBus } from './view/view-event-bus';
+import { StorableEvent } from './interfaces';
+import { IAppKnitEventBus, ViewEventBus } from './view';
+import { AggregateRootAsync } from './aggregate-root-async';
 
 @Injectable()
-export class StoreEventBus implements IEventBus {
+export class StoreEventBus implements IAppKnitEventBus {
```
Comment (bobchainer) [inline src/store-event-bus.ts]: use IEventBus as we can contribute back to original repo

Comment (sheremet) [reply src/store-event-bus.ts]: Renamed to IExtendedEventBus

**Pull request merged**